aws/table_aws_shield_protection.go

```diff
@@ -0,0 +1,276 @@
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "name",
+				Description: "The name of the protection.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "resource_arn",
+				Description: "The ARN (Amazon Resource Name) of the Amazon Web Services resource that is protected.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "arn",
+				Description: "The ARN (Amazon Resource Name) of the protection.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ProtectionArn"),
+			},
+			{
+				Name:        "resource_type",
+				Description: "The type of protected resource whose protections you want to retrieve.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("resource_type"),
+			},
+			{
+				Name:        "tags_src",
+				Description: "The list of tags associated with the protection.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listTagsForShieldProtection,
+				Transform:   transform.FromField("Tags"),
+			},
+			{
+				Name:        "health_check_ids",
+				Description: "The unique identifier (ID) for the Route 53 health check that's associated with the protection.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("HealthCheckIds").Transform(transform.EnsureStringArray),
+			},
+			{
+				Name:        "application_layer_automatic_response_configuration",
+				Description: "The automatic application layer DDoS mitigation settings for the protection. This configuration determines whether Shield Advanced automatically manages rules in the web ACL in order to respond to application layer events that Shield Advanced determines to be DDoS attacks.",
+				Type:        proto.ColumnType_JSON,
+			},
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ProtectionArn").Transform(arnToAkas),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listTagsForShieldProtection,
+				Transform:   transform.From(handleShieldProtectionTurbotTags),
+			},
+		}),
+	}
+}
+
+//// HYDRATE FUNCTIONS
+
+func listAwsShieldProtections(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create session
+	svc, err := ShieldClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_shield_protection.listAwsShieldProtections", "connection_error", err)
+		return nil, err
+	}
+
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	queryResultLimit := int32(1000)
```

Comment:
```suggestion
	queryResultLimit := int32(10000)
```

https://docs.aws.amazon.com/waf/latest/DDOSAPIReference/API_ListProtections.html